src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs

```diff
@@ -164,13 +171,19 @@ public virtual void RemoveScheme(string name)
                 if (_schemes.ContainsKey(name))
                 {
                     var scheme = _schemes[name];
-                    _requestHandlers.Remove(scheme);
-                    _schemes.Remove(name);
+                    if (_requestHandlers.Remove(scheme))
+                    {
+                        _requestHandlersCopy = _requestHandlers.ToList();
+                    }
+                    if (_schemes.Remove(name))
```

Comment:
This check is redundant because of the `_schemes.ContainsKey(name)` check inside the lock above. We already know that the scheme will be removed.